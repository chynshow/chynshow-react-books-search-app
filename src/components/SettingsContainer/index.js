import React, { useState } from 'react';
import Settings from '../Settings';
import CogSVG from '../SVG/CogSVG';

const SettingsContainer = () => {
  const [showSettings, setShowSettings] = useState(false);
  return (
    <>
      {showSettings && <Settings setShowSettings={setShowSettings} />}
      <button
        className='c-btn c-settings__btn'
        onClick={() => setShowSettings(!showSettings)}
      >
        <CogSVG />
      </button>
    </>
  );
};

export default SettingsContainer;
