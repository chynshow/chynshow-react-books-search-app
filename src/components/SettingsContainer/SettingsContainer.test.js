import React from 'react';
import SettingsContainer from './';
import { shallow } from 'enzyme';

describe('render SettingsContainer component', () => {
  it('should renders SettingsContainer component', () => {
    const c = shallow(<SettingsContainer />);
    expect(c).toMatchSnapshot();
  });
});
