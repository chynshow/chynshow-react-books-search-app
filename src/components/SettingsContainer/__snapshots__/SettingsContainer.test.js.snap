// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SettingsContainer component should renders SettingsContainer component 1`] = `
<Fragment>
  <button
    className="c-btn c-settings__btn"
    onClick={[Function]}
  >
    <CogSVG />
  </button>
</Fragment>
`;
